lib/adagrams.rb

```diff
@@ -0,0 +1,99 @@
+# This is our data structure
+pool = ["A"] * 9 + ["B"] * 2 + ["C"] * 2 + ["D"] * 4 + ["E"] * 12 + ["F"] * 2 + ["G"] * 3 + ["H"] * 2 + ["I"] * 9 + ["J"] * 1 + ["K"] * 1 + ["L"] * 4 + ["M"] * 2 + ["N"] * 6 + ["O"] * 8 + ["P"] * 2 + ["Q"] * 1 + ["R"] * 6 + ["S"] * 4 + ["T"] * 6 + ["U"] * 4 + ["V"] * 2 + ["W"] * 2 + ["X"] * 1 + ["Y"] * 2 + ["Z"] * 1
+POOL = pool
+
+# WAVE#1
+def draw_letters
+  letters_in_hand = POOL
+  return letters_in_hand.sample(10)
+end
+
+# WAVE#2:
+def uses_available_letters?(word, letters_in_hand)
+  hand_hash = letters_in_hand.uniq.map { |x| [x, letters_in_hand.count(x)] }.to_h
+  word_array = word.upcase.split("")
+  letter_check = []
+
+  word_array.map do |letter|
+    if hand_hash[letter].class == Integer
+      if hand_hash[letter] >= 1
+        hand_hash[letter] -= 1
+        letter_check << "True"
+      elsif hand_hash[letter] == 0
+        letter_check << "False"
```

Comment:
As soon as you fail to find one of the letters, you can bail! just `return false` right the heck here!